classes/CartRule.php

```diff
@@ -38,7 +38,9 @@ class CartRuleCore extends ObjectModel
     const FILTER_ACTION_ALL_NOCAP = 5;
     const BO_ORDER_CODE_PREFIX = 'BO_ORDER_';
 
-    /* This variable controls that a free gift is offered only once, even when multi-shippping is activated 
+    /* @var array $only_one_gift
```

Comment:
Don't forget that in order for a comment to be a valid PHPDoc it must start with `/**` and not `/*` (see [the proposed standard](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#3-definitions)).

Also, keep in mind that when documenting class properties, you don't need to write the property name again.

In addition, I think that for properties it is common practice to write the description before the `@var` statement.

Here's my proposal:

```php
    /**
     * This variable controls that a free gift is offered only once, even when multi-shippping is activated
     * and the same product is delivered in both addresses
     *
     * @var array
     */
```